### ISSUE TITLE
"DONE" buttons in settings don't work

### ISSUE THREAD

**Issue by dsnallfot** [dsnallfot/LoopFollowiAPS#65]
When adding remote to tab bar and pushing settings to "more" tab. Dismiss view linkage gets broken. 

UI Kit navigation defaults to adding "back" button instead = This is a minor issue, but should of course be fixed before PR to official LF repo. 

Also force unwrap crash when trying to enter infodisplaysettings. (Have fix for that in personal branch). Not 100% sure what breaks this, but work around is code to avoid force unwrapping. 

**Settings in more tab**
![image](https://github.com/dsnallfot/LoopFollowiAPS/assets/72826201/0e10974a-00c7-4710-a0b1-ff97c0779ab7)

**Default "back button" (that works). "DONE" button broken**
![image](https://github.com/dsnallfot/LoopFollowiAPS/assets/72826201/a53ffa5c-bcd6-4956-af7b-79a2fd2ddb35)

**Comment by dsnallfot** [comment]
Fixed in personal testing branch. To be PR:ed to dev-remote later.

This code:

```swift
        +++ ButtonRow() {
            $0.title = "DONE"
        }.onCellSelection { (row, arg)  in
            self.dismiss(animated:true, completion: nil)
        }
``` 

Replaced with this (allowing both modal and navigation entry points):

```swift
        +++ ButtonRow() {
            $0.title = "DONE"
        }.onCellSelection { (row, arg) in
            if let navigationController = self.navigationController {
                navigationController.popViewController(animated: true)
            } else {
                // If there's no navigation controller, dismiss the current view controller
                self.dismiss(animated: true, completion: nil)
            }
        }
```

**Comment by dsnallfot** [comment]
Fixed in PR #66
